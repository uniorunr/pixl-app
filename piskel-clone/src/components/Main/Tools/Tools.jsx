import React, { Component } from 'react';
import './Tools.scss';
import PropTypes from 'prop-types';
import ToolButton from './ToolButton/ToolButton';

class Tools extends Component {
  makeActive = ({ target }) => {
    const { updateCurrentTool } = this.props;

    const tools = [...document.querySelectorAll('.tools button')];
    tools.forEach((tool) => {
      tool.classList.remove('tools_state_active');
    });
    if (
      target.dataset.listener
      && !target.classList.contains('tools_state_active')
    ) {
      target.classList.add('tools_state_active');
      updateCurrentTool(target.id);
    } else {
      target.parentElement.classList.add('tools_state_active');
      updateCurrentTool(target.parentElement.id);
    }
  };

  handleKeyboard = ({ key }) => {
    if (key === '1') {
      this.makeActive({ target: document.querySelector('.tools__pen') });
    } else if (key === '2') {
      this.makeActive({
        target: document.querySelector('.tools__paint-bucket'),
      });
    } else if (key === '3') {
      this.makeActive({
        target: document.querySelector('.tools__choose-color'),
      });
    } else if (key === '4') {
      this.makeActive({ target: document.querySelector('.tools__eraser') });
    }
  };

  render() {
    return (
      <section className="tools-section">
        <h2 className="visually-hidden">Tools Section</h2>
        <div className="tools">
          <ToolButton
            buttonClass="tools__button tools_state_active"
            id="pen"
            iconClass="fas fa-pen"
            onClickHandler={this.makeActive}
            onKeyDownHandler={this.handleKeyboard}
          />
          <ToolButton
            buttonClass="tools__button"
            id="paint-bucket"
            iconClass="fas fa-fill-drip"
            onClickHandler={this.makeActive}
            onKeyDownHandler={this.handleKeyboard}
          />
          <ToolButton
            buttonClass="tools__choose-color"
            id="choose-color"
            iconClass="fas fa-eye-dropper"
            onClickHandler={this.makeActive}
            onKeyDownHandler={this.handleKeyboard}
          />
          <ToolButton
            buttonClass="tools__button"
            id="eraser"
            iconClass="fas fa-eraser"
            onClickHandler={this.makeActive}
            onKeyDownHandler={this.handleKeyboard}
          />
        </div>
      </section>
    );
  }
}

Tools.propTypes = {
  updateCurrentTool: PropTypes.func.isRequired,
};

export default Tools;
