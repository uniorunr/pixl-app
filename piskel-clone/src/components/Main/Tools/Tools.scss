.tools {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 30px;
  padding: 15px;
  font-weight: 700;
  font-size: 1.4rem;
  background-color: white;
  border: 2px solid black;
  cursor: context-menu;

  button {
    width: 30px;
    height: 30px;
    position: relative;
    top: 0;
    margin-bottom: 15px;
    background-color: white;
    border: 2px solid black;
    box-shadow: 2px 2px 0 black;
    cursor: pointer;
    outline: none;

    .fas {
      display: flex;
      justify-content: center;
    }
  }

  button:active {
    top: 2px;
    box-shadow: 0 0 0 black;
  }

  .tools_state_active {
    border: 3px dashed black;
  }
}
