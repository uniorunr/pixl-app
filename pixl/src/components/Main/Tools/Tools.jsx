import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './Tools.scss';
import ToolButton from './ToolButton/ToolButton';

class Tools extends Component {
  render() {
    const { toolsData, currToolId } = this.props;
    return (
      <section className="tools-section">
        <h2 className="visually-hidden">Tools Section</h2>
        <div className="tools">
          {Object.keys(toolsData).map(id => (
            <ToolButton
              buttonClass={`${toolsData[id].toolsComponentClass} ${
                currToolId === id ? 'tools_state_active' : ''
              }`}
              id={id}
              iconClass={toolsData[id].iconClass}
              key={id}
              shortcut={toolsData[id].shortcut}
            />
          ))}
        </div>
      </section>
    );
  }
}

Tools.propTypes = {
  toolsData: PropTypes.instanceOf(Object).isRequired,
  currToolId: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  currToolId: state.tools.currToolId,
  toolsData: state.tools.toolsData,
});

export default connect(mapStateToProps)(Tools);
