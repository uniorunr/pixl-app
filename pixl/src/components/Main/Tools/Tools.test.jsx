import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Tools from './Tools';
import store from '../../../stores/mainStore';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Tools />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
