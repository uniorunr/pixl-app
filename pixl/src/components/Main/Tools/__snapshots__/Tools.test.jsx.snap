// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="tools-section"
>
  <h2
    className="visually-hidden"
  >
    Tools Section
  </h2>
  <div
    className="tools"
  />
</section>
`;
